lamindb/models/collection.py

```diff
@@ -232,7 +227,7 @@ class Meta(SQLRecord.Meta, IsVersioned.Meta, TracksRun.Meta, TracksUpdates.Meta)
     @overload
     def __init__(
         self,
-        artifacts: list[Artifact],
+        artifacts: Artifact | Sequence[Artifact],
```

Comment:
I don't think we necessarily support `Sequence`.

Reply:
I think `Sequence` should be correct here and not cause issues.

```python
artifacts: Artifact | Iterable[Artifact] = (
            kwargs.pop("artifacts") if len(args) == 0 else args[0]
        )
```

used to be in the code which is likely less correct and more broad. So technically, this correction tones it down.

Reply:
Yes, `Iterable` is wrong. But `Sequence` might be incorrect too. We've only tested this with lists.